File: src/main/java/red/sukun1899/repository/TableRepository.java
package red.sukun1899.repository;

import org.apache.ibatis.annotations.Mapper;
import org.apache.ibatis.annotations.Param;
import org.springframework.stereotype.Repository;

import red.sukun1899.model.Table;

import java.util.List;
import java.util.Map;

/**
 * @author su-kun1899
 */
@Repository
@Mapper
public interface TableRepository {
  List<Table> selectAll(String schemaName);

  Table select(@Param("schemaName") String schemaName, @Param("name") String name);

  Map<String, Long> selectParentTableCountsByTableName(String schemaName);
}

File: src/main/kotlin/red/sukun1899/shishamo/repository/TableRepository.kt
package red.sukun1899.shishamo.repository

import red.sukun1899.shishamo.model.DataDefinition
import red.sukun1899.shishamo.model.ReferredTable
import red.sukun1899.shishamo.model.Table
import red.sukun1899.shishamo.model.TableDetails

/**
 * @author su-kun1899
 */
interface TableRepository {
    fun selectAll(schemaName: String): List<TableDetails>
    fun select(schemaName: String, name: String): TableDetails
    fun selectParentTableCountsByTableName(schemaName: String): Map<String, ReferredTable>
    fun selectChildTableCountsByTableName(schemaName: String): Map<String, ReferredTable>
    fun selectColumnCountsByTableName(schemaName: String): Map<String, ReferredTable>
    fun showCreateTableStatement(table: Table): DataDefinition
}
